pkg/virt-controller/services/template.go

```diff
@@ -1457,7 +1461,10 @@ func (t *templateService) VMIResourcePredicates(vmi *v1.VirtualMachineInstance,
 		vmiCPUArch = t.clusterConfig.GetClusterCPUArch()
 	}
 	memoryOverhead := GetMemoryOverhead(vmi, vmiCPUArch, t.clusterConfig.GetConfig().AdditionalGuestMemoryOverheadRatio)
-	memoryOverhead.Add(netbinding.CalculateMemoryOverhead(vmi, t.clusterConfig.GetNetworkBindings()))
+
+	if vmispec.IsVMIUsingNetworkBindingPlugins(vmi) {
```

Comment:
> IsVMIUsingNetworkBindingPlugins() was introduced in order to
not change additional unit tests which are using the real `templateService`.
Otherwise, they would break since they are not provided with the memory overhead
calculator stub.

This is not making a lot of sense to me.
You have two other alternatives:
- If not specified when creating the new virt-handler controller,  a default one can be created.
- If not specified when creating the new virt-handler controller, the condition here can check for the existence of the object and only then operate on it.

I think the 1st option above is better, as it fits with the actual production code.
When tests want to use a specific concrete implementation, it can pass it in and override the default.

Reply:
We've discussed this point offline.

I've decided to move forward with comparing the memory overhead calculator interface to nil, and performing the action if it isn't nil.

This has a disadvantage that there is currently no unit test verifying that the concrete memory overhead calculator is actually set in production - thus not necessarily adding the memory overhead.

We would probably need to add such an assertion in the e2e suite.